### ISSUE TITLE
moment().valueOf returns a value even though it's invalid

### ISSUE THREAD

**Issue by tmkx** [moment/moment-timezone#1075]
### Environment

```js
moment.tz.setDefault();
t.ok(
  Number.isNaN(moment("2023/10/06", ["H", "HH", "HH:mm"], true).valueOf()),
);

moment.tz.setDefault('America/New_York');
// not expected
t.ok(
  Number.isNaN(moment("2023/10/06", ["H", "HH", "HH:mm"], true).valueOf()),
);
```

### Issue description

when defaultZone is specified, `valueOf` will return a value even though it's invalid.

it seems to be caused by:
https://github.com/moment/moment-timezone/blob/dffed7a8a9936e5be023ca76d389c258f1746dcd/moment-timezone.js#L643-L646

a fix that works:
```diff
- if (zone && needsOffset(mom) && !mom._isUTC) { 
+ if (zone && needsOffset(mom) && !mom._isUTC && mom.isValid()) { 
    mom._d = moment.utc(mom._a)._d;
    mom.utc().add(zone.parse(mom), 'minutes');
  }
```

**Comment by sslincoco** [comment]
您的邮件已收到,我会尽快给您回复。

**Comment by gilmoreorless** [comment]
Thanks for the detailed report. I've merged #1082 which contains the fix, and added your example as a test case in [ece926a](https://github.com/moment/moment-timezone/commit/ece926a59f53c72b51e8250ef97393900c0b7227).
